### PR TITLE
Add Code Scanning to build pipeline

### PR DESCRIPTION
**Pull request by riedgar-ms**
Adding CredScan and PoliCheck, which requires a separate job since they are Windows tools

Note that while the scans run and the logs are published they have to be checked manually

### CODE REVIEW

File: azure-pipelines.yml [hunk 1]
```diff
@@ -11,24 +11,57 @@ schedules:
       - dev
   always: true
 
-pool:
-  vmImage: 'ubuntu-latest'
+# =================================================
```
Comment (romanlutz) [inline azure-pipelines.yml]: should this just go into a new file?

File: azure-pipelines.yml [hunk 2]
```diff
@@ -11,24 +11,57 @@ schedules:
...
+  - script: python -m pytest test/ --junitxml=./TEST-TEST.xml 
+    displayName: 'Run tests'
+
+  - task: PublishTestResults@2
+    displayName: 'Publish Test Results **/TEST-*.xml'
+    condition: succeededOrFailed()
+
+# -----
+
+- job: "WindowsCodeScans"
```
Comment (romanlutz) [inline azure-pipelines.yml]: I don't see a requirement to run it for every PR anywhere.

Comment (romanlutz) [reply azure-pipelines.yml]: I think it's called trigger.

Comment (riedgar-ms) [reply azure-pipelines.yml]: I'd rather do that - keep anything bad out of the codebase.

It's already flagging you for having ROC everywhere :-)

File: azure-pipelines.yml (hunk 1 above)
Comment (riedgar-ms) [reply azure-pipelines.yml]: I have to admit I wasn't sure how to do that. The name `azure-pipelines.yml` appears to be magic; I'm not sure how to have multiple pipelines defined and have them recognised by the Azure Pipelines infrastructure. I was hoping Beth would show us that.

Comment (romanlutz) [reply azure-pipelines.yml]: That's set in VSO actually.

File: azure-pipelines.yml (hunk 2 above)
Comment (romanlutz) [reply azure-pipelines.yml]: ROC is a perfectly fine acronym, and I stand by that :D 
Maybe we can add the trigger then? We can add exceptions for ROC and potential other ones

File: azure-pipelines.yml (hunk 1 above)
Comment (riedgar-ms) [reply azure-pipelines.yml]: I'm not surprised :-)

File: azure-pipelines.yml (hunk 2 above)
Comment (riedgar-ms) [reply azure-pipelines.yml]: There are definitely ways to add exceptions; just need to read up on them

Comment (riedgar-ms) [reply azure-pipelines.yml]: Have done so. There's now a separate pipeline

Comment (riedgar-ms) [reply azure-pipelines.yml]: For now, the results don't actually go anywhere, so it's not relevant.

**Comment by riedgar-ms** [comment]
Apparently should not be doing this

**Pull request closed without merging**